studioclass-frontend/src/components/PostList.js

```diff
@@ -14,19 +17,73 @@ const styles = {
   }
 }
 
-const PostList = ({ posts }) => {
+const sortPostsOn = {
+  date: 'DATE'
+}
+
+const sortDirection = {
+  ascending: 'ASC',
+  descending: 'DESC'
+}
+
+const observerOptions = {
+  threshold: 0.9
+};
+
+const FETCH_THRESHOLD = 3;
+
+const PostList = () => {
+  const { setError } = useErrorContext();
+  const { posts, fetchMore, loading } = usePosts(setError, {
+    orderBy: sortPostsOn.date,
+    orderDirection: sortDirection.descending
+  });
   const history = useHistory();
+  const observer = useRef();
 
-  const clickCard = (item) => {
+  const observerRef = useCallback(element => {
+      if (loading) return;
+      if (observer.current) observer.current.disconnect();
+
+      observer.current = new IntersectionObserver((entries) => {
+        if (entries[0].isIntersecting) {
+          fetchMore()
+        }
+      }, observerOptions);
+
+      if (element) {
+        observer.current.observe(element)
+      }  
+  }, [loading, fetchMore])
+  
+
+  const clickCard = useCallback((item) => {
     history.push(`/post/${item.id}`);
-  }
+  }, [history])
+
+  console.log('postlist posts', posts);
```

Comment:
Remove console. On the front end, they can expose your apis and internal data to would be hackers.